tests/integration/test_serve_agent.py

```diff
@@ -0,0 +1,56 @@
+import multiprocessing
+import time
+from typing import Any
+from uuid import uuid4
+
+import httpx
+import pytest
+from a2a.client import A2AClient
+from a2a.types import MessageSendParams, SendMessageRequest
+
+# Import your agent and config
+from any_agent import AgentConfig, AnyAgent
+from any_agent.config import ServingConfig
+
+SERVER_PORT = 5000
+
+
+def run_agent():
+    agent = AnyAgent.create(
+        "openai",
+        AgentConfig(
+            model_id="gpt-4.1-mini",
+            description="I'm an agent to help with booking airbnbs",
+        ),
+    )
+    agent.serve(serving_config=ServingConfig(port=SERVER_PORT))
+
+
+@pytest.mark.asyncio
+async def test_agent_serving_and_communication():
+    # Start the agent in a subprocess
+    proc = multiprocessing.Process(target=run_agent, daemon=True)
+    proc.start()
+    time.sleep(5)
```

Comment:
I know I need to tweak this so that it's not a sleep, but runs health checks

Reply:
TBF the client will (or at least can) be reconfigured for retries so it's not so important. If the uvicorn server were available and in the same memory space we could just check https://github.com/encode/uvicorn/blob/fa3d9d27dbe7ecbc68e0992309d0ba49ead5640b/uvicorn/server.py#L57 and then shut it down with https://github.com/encode/uvicorn/blob/fa3d9d27dbe7ecbc68e0992309d0ba49ead5640b/uvicorn/server.py#L262 .